config/config.go

```diff
@@ -0,0 +1,628 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package config
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/state/pruner"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/ChainSafe/gossamer/lib/os"
+	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
+)
+
+const (
+	// uint32Max is the maximum value of a uint32
+	uint32Max = ^uint32(0)
+	// defaultChainSpecFile is the default genesis file
+	defaultChainSpecFile = "chain-spec-raw.json"
+	// defaultBasePath is the default base path
+	defaultBasePath = "~/.gossamer"
+	// DefaultLogLevel is the default log level
+	DefaultLogLevel = "info"
+	// DefaultPrometheusPort is the default prometheus port
+	DefaultPrometheusPort = uint32(9876)
+	// DefaultRetainBlocks is the default number of blocks to retain
+	DefaultRetainBlocks = 512
+	// DefaultPruning is the default pruning strategy
+	DefaultPruning = pruner.Archive
+
+	// DefaultRole is the default node role
+	DefaultRole = common.AuthorityRole
+	// DefaultWasmInterpreter is the default wasm interpreter
+	DefaultWasmInterpreter = wasmer.Name
+
+	// DefaultNetworkPort is the default network port
+	DefaultNetworkPort = 7001
+	// DefaultDiscoveryInterval is the default discovery interval
+	DefaultDiscoveryInterval = 10 * time.Second
+	// DefaultMinPeers is the default minimum number of peers
+	DefaultMinPeers = 0
+	// DefaultMaxPeers is the default maximum number of peers
+	DefaultMaxPeers = 0
```

Comment:
I feel like maybe we want default max peers?

Reply:
good catch!

This gets updated in `dot/network/config.go` to 5 and 50 for if this is set to 0.  Maybe it's better to mirror the same the same values in here.  We can get rid of that checking logic in `dot/network` at a later date.

Reply:
I am not sure if we can do that. The same config is being used by both `network` and `sync` package, but it only sets the default ( 5 and 50 ) in `network` but not in the `sync`.

Reply:
I've created an issue for the config here #3240

Reply:
Maybe I am misunderstanding, but wouldn't that be a good reason to set default values in this file so that the values are consistent in network and sync services?

Reply:
Maybe you're right. Since it was setting the default value only for the `network` service, it made me think we could have a separate config for `sync` service

Reply:
Yea I think using the same values for both `network` and `sync` make sense.

Reply:
The `sync` service doesn't start until it is connected to `DefaultMinPeers` which is making the tests fail.
I've set only the `DefaultMaxPeers`